src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs

```diff
@@ -180,18 +191,24 @@ internal string AadAuthorityV2
 
             try
             {
-                BaseConfigurationManager effectiveConfigurationManager = GetEffectiveConfigurationManager(securityToken);
-                if (validationParameters.RefreshBeforeValidation)
-                    effectiveConfigurationManager.RequestRefresh();
+                var isV2Issuer = IsV2Issuer(securityToken);
 
-                BaseConfiguration configuration = await effectiveConfigurationManager.GetBaseConfigurationAsync(CancellationToken.None).ConfigureAwait(false);
-                string aadIssuer = configuration.Issuer;
+                var effectiveConfigurationManager = GetEffectiveConfigurationManager(isV2Issuer);
```

Comment:
Could we just pass the token to GetEffectiveConfigurationManager and bypass the extra call to IsV2Issuer?

Reply:
we seem to be missing the logic: 
 if (validationParameters.RefreshBeforeValidation)
                    effectiveConfigurationManager.RequestRefresh();

Reply:
GetEffectiveConfigurationManager does not call IsV2Issuer nor uses the token. It only needs information on the issuer version and it's passed as bool.